src/services/dbgap_telemetry_report.py

```diff
@@ -0,0 +1,49 @@
+import requests
+import xmltodict
+
+class SampleNotFoundError(Exception):
+    """Exception raised when a sample is not found."""
+    def __init__(self, alias):
+        self.alias = alias
+        super().__init__(f"Sample with alias '{alias}' not found")
+
+class DbgapTelemetryWrapper:
+    def __init__(self, phs_id=None):
+        self.endpoint = f"https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetSampleStatus.cgi?rettype=xml&study_id={phs_id}"
+        self.phs_id = phs_id
+
+    def call_telemetry_report(self):
+        """Example xml - https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetSampleStatus.cgi?rettype=xml&study_id=phs000452"""
+        response = requests.get(
+            self.endpoint, 
+            headers={"Content-Type": "application/json"}
+        )
+        status_code = response.status_code
+
+        return xmltodict.parse(response.text)
+
+    def get_sample_status(self, alias, data_type):
+        try:
+            telemetry_data = self.call_telemetry_report()
+
+            study_data = telemetry_data["DbGap"]["Study"]
+            sample_list = study_data["SampleList"]["Sample"]
+
+            for sample in sample_list:
+                if sample.get("@submitted_sample_id") == alias:
+                    sra_data = sample["SRAData"]
+                    sra_sample_stats = sra_data["Stats"]
+
+                    if isinstance(sra_sample_stats, list):
+                        for stat in sra_sample_stats:
+                            if stat.get("@experiment_type") == data_type:
+                                return stat["@status"]
+                    elif isinstance(sra_sample_stats, dict):
+                        return sra_sample_stats["@status"]
+            
+            raise SampleNotFoundError(alias)
```

Comment:
Maybe I'm misunderstanding, but doesn't this need to be indented one more? I think we want to raise this exception only if line 33 doesn't evaluate as true, correct?

Reply:
So the for loop on line 32 goes through each sample returned from dbgap. I was thinking that this exception would be thrown if we loop through each sample and nothing is returned

Reply:
Yes totally, ignore me! I was wrong about this